DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,283 @@
+- harvester/terraform-provider-harvester
+  - Terraform provider for Harvester.
+- harvester/node-disk-manager
+  - Provides an automated way to add storage to Longhorn as Harvester's backend storage, including multipath support.
+- harvester/harvester-ui-extension
+  - Harvester Dashboard UI. For more details, see the [official documentation](https://docs.harvesterhci.io/v1.6/rancher/harvester-ui-extension/).
+
+
+### Add-ons
+
+### Built-in
+
+- harvester/pcidevices
+  - PCI, USB, GPU, and vGPU device passthrough.
+- harvester/nvidia-driver-toolkit
+  - Enable vGPU devices and assign them to Harvester virtual machines. It's used with harvester/pcidevices together.
+- harvester/rancher-logging
+  - Collect versatile logs, events, and audits from the Harvester cluster and route them to many kinds of servers based on flows.
+- harvester/rancher-monitoring
+  - Collect Harvester cluster and virtual machine metrics, view them on the embedded dashboard, and send alert(s) to remote servers.
+- harvester/vm-import-controller
+  - Helps migrate VM workloads from external clusters to an existing Harvester cluster. It currently supports VMware and OpenStack.
+
+#### Experimental
+
+- harvester/experimental-addons/harvester-csi-driver-lvm
+  - CSI driver that supports local path provisioning through LVM
+- harvester/experimental-addons/vm-dhcp-controller
+  - A managed DHCP service for virtual machines running on Harvester.
+- harvester/experimental-addons/rancher-vcluster
+  - Runs Rancher Manager as a workload on the underlying Harvester cluster, implemented using vcluster. It's one way to start a Rancher service and use [Rancher Integration](https://docs.harvesterhci.io/v1.5/rancher/rancher-integration) with Harvester.
+
+
+### Rancher-Related
+
+- harvester/docker-machine-driver-harvester
+  - Use Harvester as a cloud provider to provision guest clusters in Rancher. The related UI is located [here](https://github.com/rancher/dashboard/tree/master/pkg/harvester-manager).
+
+### Most Closely Integrated Upstream Repositories
+
+- longhorn/longhorn
+  - Harvester uses Longhorn for virtual machine and node volumes.
+- kubevirt/kubevirt
+  - Harvester uses KubeVirt to provide virtualization.
+
+## Development
+
+### Prerequisites
+
+Before you start, ensure the following are installed on your development machine:
+
+- OS: Linux or macOS (Linux is required to build ISOs; macOS is fine for day-to-day development).
+- Go: See go.mod and use that major.minor version to avoid module/tooling mismatches.
+- Docker-compatible container engine: required by Dapper for builds (Docker Engine or Docker Desktop; Rancher Desktop or Colima also work).
+- Make and Git: used to invoke build scripts and manage source.
+
+Notes:
+- Dapper is downloaded automatically by make; you just need a working container engine.
+- To push locally built images, you’ll set environment variables such as REPO, PUSH, and USE_LOCAL_IMAGES (see [Testing and Building](#testing-and-building)).
```

Comment:
Common environment variables are as follows:
- REPO: target repository
- TAG: target tag (for image)
- PUSH: push to target repository
- RKE2_IMAGE_REPO: use for ISO builder
- USE_LOCAL_IMAGES: prevent to pulling image from remote